KachnaOnline.Business/Services/BoardGamesNotifications/NotificationHandlers/MailBoardGamesNotificationHandler.cs

```diff
@@ -0,0 +1,195 @@
+// MailBoardGamesNotificationHandler.cs
+// Author: František Nečas
+
+using System.Net;
+using System.Net.Mail;
+using System.Threading.Tasks;
+using KachnaOnline.Business.Configuration;
+using KachnaOnline.Business.Constants;
+using KachnaOnline.Business.Exceptions.BoardGames;
+using KachnaOnline.Business.Models.Users;
+using KachnaOnline.Business.Services.Abstractions;
+using KachnaOnline.Business.Services.BoardGamesNotifications.Abstractions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace KachnaOnline.Business.Services.BoardGamesNotifications.NotificationHandlers
+{
+    /// <summary>
+    /// Implements a handler sending messages via e-mail.
+    /// </summary>
+    public class MailBoardGamesNotificationHandler : IBoardGamesNotificationHandler
+    {
+        private readonly IOptionsMonitor<SmtpOptions> _smtpOptionsMonitor;
+        private readonly ILogger<MailBoardGamesNotificationHandler> _logger;
+        private readonly IBoardGamesService _boardGamesService;
+        private readonly IUserService _userService;
+
+        public MailBoardGamesNotificationHandler(IOptionsMonitor<SmtpOptions> smtpOptionsMonitor,
+            ILogger<MailBoardGamesNotificationHandler> logger, IBoardGamesService boardGamesService,
+            IUserService userService)
+        {
+            _smtpOptionsMonitor = smtpOptionsMonitor;
+            _logger = logger;
+            _boardGamesService = boardGamesService;
+            _userService = userService;
+        }
+
+        /// <inheritdoc />
+        /// <remarks>
+        /// Does nothing.
+        /// </remarks>
+        public Task PerformReservationCreated(int reservationId)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        /// <remarks>
+        /// Does nothing.
+        /// </remarks>
+        public Task PerformReservationFullyAssigned(int reservationId)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        /// <remarks>
+        /// Does nothing.
+        /// </remarks>
+        public Task PerformReservationItemExtensionRequest(int itemId)
+        {
+            return Task.CompletedTask;
+        }
```

Comment:
Because the class implements a common interface for the handlers so it must be implemented. Discord handler implements logic in this method so in order to have a common interface, this must implement it too. And the service running these handlers goes through all of the registered handlers and tries to run the appropriate method. So if it threw a `NotImplementedError`, we would have to catch it there, which is far from ideal and I'd rather let it do nothing and just return that the task is completed.